src/main/java/com/example/chat/groupchatbackend/controllers/ConversationController.java

```diff
@@ -73,5 +74,30 @@ public ResponseEntity getAllChannelMembers(@PathVariable int id) {
                 .status(HttpStatus.BAD_REQUEST)
                 .body("not channel");
     }
+
+    @PutMapping("/channels/{id}/join")
+    @Transactional
+    public ResponseEntity joinChannel(@PathVariable int id){
+        Conversation conversation = conversationsRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("conversation doesn't exist"));
+        User currentUser = userContext.getCurrentUser();
+
+        if(conversation.getConversationType().equals(ConversationType.CHANNEL) && currentUser != null){
+            if(!conversation.checkIfUserIsInConversation(currentUser)) {
+                conversation.getUsers().add(currentUser);
+                return new ResponseEntity(HttpStatus.OK);
+            }
+            else {
+                return ResponseEntity
```

Comment:
should return OK if a user was newly added or was already in the conversation. ask me for details why if you would like :)